libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefresherListener.kt

```diff
@@ -1,5 +1,10 @@
 package com.mapbox.navigation.core.routerefresh
 
 internal fun interface RouteRefresherListener {
-    fun onRoutesRefreshed(result: RouteRefresherResult)
+    fun onRoutesRefreshed(result: RoutesRefresherResult)
+}
+
+internal fun interface RoutesRefreshAttemptListener {
+
+    fun onRoutesRefreshAttemptFinished(result: RoutesRefresherResult)
```

Comment:
Why did you split listeners in two interfaces? 🤔
Did you want to fire invalidate callback earlier than route refresh state changes?

Reply:
As the name suggests, this one is invoked after every attempt, while `onRoutesRefreshed` in case of failure - only after 3 attempts. In case we receive an error on the first attempt with refresh_ttl = 0, it makes sense to notify the user right away.